pkg/pgmodel/migrations/sql/idempotent/base.sql

```diff
@@ -1196,6 +1212,19 @@ BEGIN
         AND c.compressed_chunk_id IS NOT NULL
         ORDER BY ds.range_start
     LOOP
+
+        --lock the chunk exclusive.
+        EXECUTE format('LOCK %I.%I;', chunk_row.schema_name, chunk_row.table_name);
+        --double check it's still compressed.
+        PERFORM c.*
+        FROM _timescaledb_catalog.chunk c
+        WHERE c.id = chunk_row.id AND c.compressed_chunk_id IS NOT NULL;
+
+        IF NOT FOUND THEN
```

Comment:
It might be cleaner to just do the notice and decompression in an if block rather than having a commit/continue here, but I don't really mind it either way.